Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -241,6 +260,50 @@ private void OnPlayerSpawn(PlayerSpawningEvent ev)
         }
     }
 
+    private bool TryTeleportToMapSpawn(EntityUid player, EntityUid stationId, TransformComponent? transform = null)
+    {
+        if (!Resolve(player, ref transform))
+            return false;
+
+        var points = EntityQueryEnumerator<SpawnPointComponent, TransformComponent>();
+        var possiblePositions = new List<EntityCoordinates>();
+        var passengerPositions = new List<EntityCoordinates>();
```

Comment:
ValueList?

Reply:
Wow, I didn't know we had that.

Reply:
Now require https://github.com/space-wizards/RobustToolbox/pull/4257 for Pick(ValueList)